Gems/FastNoise/Code/Source/FastNoiseGradientComponent.cpp

```diff
@@ -294,13 +298,20 @@ namespace FastNoiseGem
         return false;
     }
 
+    void FastNoiseGradientComponent::OnGradientTransformChanged(const GradientSignal::GradientTransform& newTransform)
+    {
+        AZStd::unique_lock<decltype(m_transformMutex)> lock(m_transformMutex);
+        m_gradientTransform = newTransform;
+    }
+
     float FastNoiseGradientComponent::GetValue(const GradientSignal::GradientSampleParams& sampleParams) const
     {
+        AZStd::shared_lock<decltype(m_transformMutex)> lock(m_transformMutex);
+
         AZ::Vector3 uvw = sampleParams.m_position;
 
         bool wasPointRejected = false;
-        GradientSignal::GradientTransformRequestBus::Event(
-            GetEntityId(), &GradientSignal::GradientTransformRequestBus::Events::TransformPositionToUVW, sampleParams.m_position, uvw, wasPointRejected);
+        m_gradientTransform.TransformPositionToUVW(sampleParams.m_position, uvw, wasPointRejected);
```

Comment:
You could reduce the scope of the shared lock a bit instead of covering the whole function since you really only care about m_gradientTransform changing while you're using it.
```
AZ::Vector3 uvw = sampleParams.m_position;

bool wasPointRejected = false;
{
    AZStd::shared_lock<decltype(m_transformMutex)> lock(m_transformMutex);
    m_gradientTransform.TransformPositionToUVW(sampleParams.m_position, uvw, wasPointRejected);
}
````